pkg/cmd/roachtest/cluster.go

```diff
@@ -1086,6 +1086,37 @@ func (c *cluster) FetchDmesg(ctx context.Context) error {
 	})
 }
 
+// FetchJournalctl grabs the journalctl logs if possible. This requires being
+// able to run `sudo journalctl` on the remote nodes.
+func (c *cluster) FetchJournalctl(ctx context.Context) error {
+	if c.nodes == 0 || c.isLocal() {
+		// No nodes can happen during unit tests and implies nothing to do.
+		// Also, don't grab dmesg on local runs.
+		return nil
+	}
+
+	c.l.Printf("fetching journalctl\n")
+	c.status("fetching journalctl")
+
+	// Don't hang forever.
+	return contextutil.RunWithTimeout(ctx, "journalctl", 20*time.Second, func(ctx context.Context) error {
+		const name = "journalctl.txt"
+		path := filepath.Join(c.t.ArtifactsDir(), name)
+		if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
+			return err
+		}
+		if err := execCmd(
+			ctx, c.l, roachprod, "ssh", c.name, "--",
+			"/bin/bash", "-c", "'sudo journalctl > "+name+"'", /* src */
```

Comment:
I assume you tested this.

Reply:
(I also know that it's mostly copy pasta)

Reply:
Yes, I tested it and looked at the journalctl logs.